### PR TITLE
Feature/v2 updates

### PR DESCRIPTION
**Pull request by fivetran-jamie**
Pull Request
**Are you a current Fivetran customer?** 

fivetran made PR

**What change(s) does this PR introduce?** 

- adds line item, campaign, and account report tables
- adds `tweet` table as well
- incorporates passthrough columns into report tables
- adds schema tests to all staging models 
- aligns promoted_tweet_report model with the other report models
- updates README
- documents new tables and columns
- incorporate `identifier` variables
- TODO AFTER APPROVING: regen docs

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [x] No 

**How did you test the PR changes?** 

 

- [x] CircleCi  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🍏 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/src_twitter_ads.yml
Comment (fivetran-joemarkiewicz) [inline models/src_twitter_ads.yml:233]: I am a bit confused by this report. Are line items in Twitter essentially the equivalent of Ad Groups? 

I see in the [Ad Reporting Discovery doc](https://docs.google.com/document/d/1AvYasVocqqQ7h8C6WQ3ivslA707t4UTSzRwB3hTJYBw/edit#heading=h.ctmks52hja9n) you mention the hierarchy is 
```txt
Account > campaigns > ad groups > promoted tweets
```
Do line items fit into the ad group section?

File: models/stg_twitter_ads.yml
Comment (fivetran-joemarkiewicz) [inline models/stg_twitter_ads.yml:193]: It looks like `_fivetran_synced` is documented, but it isn't actually used in the staging model. Can we either remove it from the docs or add it to the staging model. 
image

File: dbt_project.yml
Comment (fivetran-joemarkiewicz) [inline dbt_project.yml:7]: Strangely enough when I run the package **all** my models are `views`. Can we define the default materialization here as well.

File: models/src_twitter_ads.yml
Comment (fivetran-joemarkiewicz) [inline models/src_twitter_ads.yml:257]: Why do we use `config` here, but in other packages we use `meta`?
https://github.com/fivetran/dbt_zendesk_source/blob/aef4f3e6d436b4d5f52ecac81c8beb71951289b0/models/src_zendesk.yml#L190-L193

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie thanks for this PR! There definitely was a lot added to this package, thanks for working though all of these updates! 🙌 

The PR looks great, I just had a few inline comments to be addressed as well as a few thoughts I would like your insight on before approving. See inline comments for more details.

**Comment by fivetran-joemarkiewicz** [comment]
@fivetran-joemarkiewicz to change variable to be `pinterest_ads`

File: macros/staging_columns.sql
Comment (fivetran-jamie) [reply macros/staging_columns.sql:15]: yeah i think these are fine to move to the staging models -- this format is just a vestige of how we used to only alias stuff in the macros

File: models/tmp/stg_twitter_ads__line_item_keywords_report_tmp.sql
Comment (fivetran-jamie) [reply models/tmp/stg_twitter_ads__line_item_keywords_report_tmp.sql:1]: do you think that should fall under `Additional configurations` or should it be a required step in the README?

File: models/src_twitter_ads.yml
Comment (fivetran-jamie) [reply models/src_twitter_ads.yml:233]: yep line items = ad groups

File: models/stg_twitter_ads.yml
Comment (fivetran-jamie) [reply models/stg_twitter_ads.yml:193]: removed

File: models/src_twitter_ads.yml
Comment (fivetran-jamie) [reply models/src_twitter_ads.yml:7]: added same kind of freshness test

File: models/src_twitter_ads.yml
Comment (fivetran-jamie) [reply models/src_twitter_ads.yml:257]: presuming we're good to use config here now that we've tested it out?

File: dbt_project.yml
Comment (fivetran-jamie) [reply dbt_project.yml:7]: are `views` the default materialization if you don't specify? i'll make `_tmp` models views and non-tmp ones tables

File: models/src_twitter_ads.yml
Comment (fivetran-jamie) [reply models/src_twitter_ads.yml:7]: huh interesting -- our own keywords data is pretty stale. if i were a user of this package, i think i would want to know that and then would exclude keywords using the `twitter_ads__using_keywords` var (or figure out why keywords are not being tracked and start tracking them).. right?

File: macros/staging_columns.sql
Comment (fivetran-jamie) [reply macros/staging_columns.sql]: switched to renaming in the model files 🤠

File: models/tmp/stg_twitter_ads__line_item_keywords_report_tmp.sql
Comment (fivetran-joemarkiewicz) [reply models/tmp/stg_twitter_ads__line_item_keywords_report_tmp.sql:1]: This should be a required step since it technically can cause a run to fail from the start if a user doesn't have the source table.

File: models/src_twitter_ads.yml
Comment (fivetran-joemarkiewicz) [reply models/src_twitter_ads.yml:264]: Yup all good!

File: dbt_project.yml
Comment (fivetran-joemarkiewicz) [reply dbt_project.yml:10]: Turns out the default is a [view](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/materializations#configuring-materializations). Thanks for updating!

File: models/src_twitter_ads.yml
Comment (fivetran-joemarkiewicz) [reply models/src_twitter_ads.yml:14]: I would agree that this would seem like the logically progression.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie updates look great!! Only request is to move the disabling docs to the required section of the README opposed to the additional.

Once we make that update we will be good to hold this PR until all others are ready to merge.

**Comment by fivetran-joemarkiewicz** [comment]
@fivetran-jamie one last thing I noticed was that the date_day field for the reports seemed to report on different hours each day. Is this regular? I would imagine the date_day field to simple display the date and not an hour of the day?

**Comment by fivetran-jamie** [comment]
> @fivetran-jamie one last thing I noticed was that the date_day field for the reports seemed to report on different hours each day. Is this regular? I would imagine the date_day field to simple display the date and not an hour of the day?

good catch -- yeah i think date_trunc'ing `date_day` in the staging models is the best move

**Pull request merged**
